Sources/Apodini/Semantic Model Builder/SemanticModelBuilder.swift

```diff
@@ -70,78 +59,16 @@ class SemanticModelBuilder: InterfaceExporterVisitor {
-            guard let self = self else {
-                return
-            }
-            
-            let paths = context.get(valueFor: PathComponentContextKey.self)
-            
-
-            let partialCandidates = context.get(valueFor: RelationshipSourceCandidateContextKey.self)
-            let relationshipSources = context.get(valueFor: RelationshipSourceContextKey.self)
-            let relationshipDestinations = context.get(valueFor: RelationshipDestinationContextKey.self)
-
-            var endpoint = Endpoint(
-                handler: handler,
-                blackboard: localBlackboard
-            )
-
-            self.webService.addEndpoint(&endpoint, at: paths)
-            // The `ReferenceModule` and `EndpointPathModule` cannot be implemented using one of the standard
-            // `KnowledgeSource` protocols as they depend on the `WebServiceModel`. This should change
-            // once the latter was ported to the Blackboard-Pattern.
-            endpoint[ReferenceModule.self].inject(reference: endpoint.reference)
-            endpoint[EndpointPathModule.self].inject(absolutePath: endpoint.absolutePath)
-            
-
-            // calling the addEndpoint first, triggers the Operation uniqueness check
-            // and we will have less problems with that in the TypeIndex Builder and Relationship Builders.
-            // Additionally, addEndpoint may cause a insertion of a additional path parameter,
-            // which makes it necessary to be called before any operation relying on the path of the Endpoint.
-
-            self.relationshipBuilder.collect(
-                endpoint: endpoint,
-                candidates: partialCandidates,
-                sources: relationshipSources,
-                destinations: relationshipDestinations
-            )
-            
-            let content = endpoint[HandleReturnType.self].type
-            let reference = endpoint[ReferenceModule.self].reference
-            let markedDefault = endpoint[Context.self].get(valueFor: DefaultRelationshipContextKey.self) != nil
-            let pathParameters = endpoint[EndpointPathModule.self].absolutePath.listPathParameters()
-            let operation = endpoint[Operation.self]
-            
-            self.typeIndexBuilder.indexContentType(content: content,
-                                                   reference: reference,
-                                                   markedDefault: markedDefault,
-                                                   pathParameters: pathParameters,
-                                                   operation: operation)
-        }
+        // We first only build the blackboards and the `Endpoint`. The validation and exporting is done at the
+        // beginning of `finishedRegistration`. This way `.global` `KnowledgeSource`s get a complete view of
+        // the web service even when accessed from an `Endpoint`.
+        endpointsToExport.append(Endpoint(
```

Comment:
I really like how the `SemanticModelBuilder` has trimmed down now. Great job with the `Blackboard` pattern

Reply:
Thanks!